src/h5web/explorer/utils.ts

```diff
@@ -1,57 +1,126 @@
 import { nanoid } from 'nanoid';
-import type { TreeNode } from './models';
 import {
   HDF5Collection,
-  HDF5Link,
   HDF5LinkClass,
   HDF5RootLink,
   HDF5Metadata,
+  MyHDF5Entity,
+  MyHDF5Group,
+  MyHDF5Dataset,
+  MyHDF5Datatype,
+  HDF5HardLink,
+  MyHDF5EntityKind,
 } from '../providers/models';
-import { isReachable } from '../providers/utils';
-
-function buildTreeNode(
-  metadata: HDF5Metadata,
-  link: HDF5Link,
-  parents: HDF5Link[]
-): TreeNode<HDF5Link> {
-  const group =
-    isReachable(link) && link.collection === HDF5Collection.Groups
-      ? metadata.groups[link.id]
-      : undefined;
+import { isMyGroup, isReachable } from '../providers/utils';
 
+function prepareEntity(
+  link: HDF5HardLink | HDF5RootLink,
+  parents: MyHDF5Group[]
+): Omit<MyHDF5Entity, 'kind'> {
   return {
     uid: nanoid(),
-    label: link.title,
-    data: link,
+    id: link.id,
```

Comment:
Would we be able to merge `id` and `uid` later ?

Reply:
Well, it's the usual conundrum of whether the same entity can be referenced from multiple links, but yes, we can definitely expose a single ID in the structure and let the providers either:

- use the IDs returned by the server if they are guaranteed to be unique; or
- generate unique IDs dynamically.

If a provider ends up generating unique IDs, it will have to keep a mapping of the generated IDs to the raw IDs for the fetching of dataset values to keep working.